plugins/woocommerce/tests/php/src/Internal/Admin/ReviewsTest.php

```diff
@@ -473,4 +476,74 @@ public function provider_display_notices() : Generator {
 		];
 	}
 
+	/**
+	 * Tests scenarios that should return false.
+	 *
+	 * @covers       \Automattic\WooCommerce\Internal\Admin\Reviews::is_review_or_reply()
+	 * @dataProvider provider_is_review_or_reply
+	 *
+	 * @param WP_Comment|array|null $object   Object to pass in to the method.
+	 * @param bool                  $expected Expected result.
+	 * @return void
+	 * @throws ReflectionException If the method doesn't exist.
+	 */
+	public function test_is_not_review_or_reply( $object, bool $expected ) : void {
+		$reviews = new Reviews();
+		$method  = ( new ReflectionClass( $reviews ) )->getMethod( 'is_review_or_reply' );
+		$method->setAccessible( true );
+
+		$this->assertSame( $expected, $method->invoke( $reviews, $object ) );
+	}
+
+	/** @see test_is_not_review_or_reply */
+	public function provider_is_review_or_reply(): Generator {
+		yield 'null object' => [ null, false ];
+		yield 'invalid array' => [ [ 'data' ], false ];
+	}
+
+	/**
+	 * Tests different cases that require factories.
+	 *
+	 * @covers \Automattic\WooCommerce\Internal\Admin\Reviews::is_review_or_reply()
+	 *
+	 * @return void
+	 * @throws ReflectionException If the method doesn't exist.
+	 */
+	public function test_is_review_or_reply_with_comment_object() : void {
```

Comment:
This is in a separate test instead of using the data provider above, because for some reason one of the cases using a factory was failing for me when run via GH actions. (Passed for me locally, which made it hard to debug...) I'm guessing it had something to do with using the factories inside a data provider. 🤷